cmd/environment.go

```diff
@@ -52,6 +37,110 @@ var deleteEnvCmd = &cobra.Command{
 	},
 }
 
+var updateEnvironmentCmd = &cobra.Command{
+	Use:     "environment",
+	Aliases: []string{"e"},
+	Short:   "Update an environment",
+	PreRunE: func(_ *cobra.Command, _ []string) error {
+		return validateTokenE(lagoonCLIConfig.Current)
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		debug, err := cmd.Flags().GetBool("debug")
+		if err != nil {
+			return err
+		}
+		deployBaseRef, err := cmd.Flags().GetString("deploy-base-ref")
+		if err != nil {
+			return err
+		}
+		deployHeadRef, err := cmd.Flags().GetString("deploy-head-ref")
+		if err != nil {
+			return err
+		}
+		namespace, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			return err
+		}
+		route, err := cmd.Flags().GetString("route")
+		if err != nil {
+			return err
+		}
+		routes, err := cmd.Flags().GetString("routes")
+		if err != nil {
+			return err
+		}
+		environmentType, err := cmd.Flags().GetString("environment-type")
+		if err != nil {
+			return err
+		}
+		deployT, err := cmd.Flags().GetString("deploy-type")
+		if err != nil {
+			return err
+		}
+		autoIdle, err := cmd.Flags().GetUint("auto-idle")
+		if err != nil {
+			return err
+		}
+		openShift, err := cmd.Flags().GetUint("deploy-target")
+		if err != nil {
+			return err
+		}
+		deployTitle, err := cmd.Flags().GetString("deploy-title")
+		if err != nil {
+			return err
+		}
+
+		if cmdProjectName == "" || cmdProjectEnvironment == "" {
+			fmt.Println("Missing arguments: Project name or environment name is not defined")
+			cmd.Help()
+			os.Exit(1)
+		}
+
+		current := lagoonCLIConfig.Current
+		token := lagoonCLIConfig.Lagoons[current].Token
+		lc := lclient.New(
+			lagoonCLIConfig.Lagoons[current].GraphQL,
+			lagoonCLIVersion,
+			&token,
+			debug)
+		project, err := l.GetMinimalProjectByName(context.TODO(), cmdProjectName, lc)
+		environment, err := l.GetEnvironmentByName(context.TODO(), cmdProjectEnvironment, project.ID, lc)
+		handleError(err)
+
+		environmentFlags := s.UpdateEnvironmentPatchInput{
+			ProjectID:            &project.ID,
+			DeployBaseRef:        &deployBaseRef,
+			DeployHeadRef:        &deployHeadRef,
+			OpenshiftProjectName: &namespace,
+			Route:                &route,
+			Routes:               &routes,
+			AutoIdle:             &autoIdle,
+			DeployTitle:          &deployTitle,
+			Openshift:            &openShift,
+		}
+		if environmentType != "" {
```

Comment:
if you do `envType := s.EnvType(strings.ToUpper(environmentType)) then the user can just enter `development` in the command line.

But what would be good is that you check that what the value the user has provided is actually a valid EnvType and return an error if it isn't. Currently I could enter whatever value I like in the field and it will get converted to the `EnvType` type, but it may not be valid. Then the CLI will perform the query against the API and the API will throw an error that the type provided is wrong.

A helper function in `machinery` would probably good for this, something like https://go.dev/play/p/zx6_6W264NN
And a similar function for `DeployType`. 

But for now just `strings.ToUpper(val)` when you pass it to the EnvType, then we can get this command added to the CLI quickly. Then you can work on the other helper stuff in a separate PR in machinery, and refactor the CLI later on.